mycroft_bus_client/client/client.py

```diff
@@ -82,6 +83,110 @@ def wait(self, timeout=3.0):
         return self.received_msg
 
 
+class MessageCollector:
+    """Collect multiple response.
+
+    This class encapsulates the logic for collecting messages from
+    multiple handlers returning the list of all answers.
+
+    Argunments:
+        bus: Bus to check for messages on
+        message (Message): message to send
+        min_timeout (int/float): Minimum time to wait for a response
+        max_timeout (int/float): Maximum allowed time to wait for an answer
+        direct_return_func (callable): Optional function for allowing an
+            early return (not all registered handlers need to respond)
```

Comment:
Is this equivalent to a validator type function, eg something like:
```python
def good_enough_response(message):
  return message.data['confidence'] > 0.9
```

So by default we wait for every handler that acknowledges, but the caller can provide a shortcut?

Sidenote - looks like it's `direct_return_func` in the docstring, and `direct_return` in the args

Reply:
Exactly so.

Reply:
Will fix the mismatch here